File: Design Patterns/basic-patterns-java/command/ConcreteCommand.java
package command;

import javafx.scene.paint.Color;
import javafx.scene.shape.Circle;

public class ConcreteCommand implements Command {

    private final Circle circle;
    private Color previousColor;

    public ConcreteCommand(Circle circle) {
        this.circle = circle;
    }

    @Override
    public void execute() {
        previousColor = (Color) circle.getFill();
        circle.setFill(Color.color(Math.random(), Math.random(), Math.random()));
    }

    @Override
    public void undo() {
        circle.setFill(previousColor);
    }

}

File: Design Patterns/basic-patterns-java-kotlin/command/ConcreteCommand.kt
package command

import javafx.scene.paint.Color
import javafx.scene.shape.Circle

class ConcreteCommand(private val circle: Circle) : Command {
    private var previousColor: Color? = null
    override fun execute() {
        previousColor = circle.fill as Color
        circle.fill = Color.color(Math.random(), Math.random(), Math.random())
    }

    override fun undo() {
        circle.fill = previousColor
    }
}